libs/sim/simulator.rs

```diff
@@ -201,6 +209,14 @@ impl<'a> Simulator {
                 }
             }
         }
+        if self.record_network_data {
+            log::info!(
```

Comment:
I am not sure if you want to collect data using a log. Instead maintain a csv for this that collects the information. Then you can also easily plot it. Although I am not sure what you are planning to use this for.